src/postgraphile-upsert.ts

```diff
@@ -161,6 +163,51 @@ function createUpsertField({
     }
   );
 
+  const DoUpdateFieldType = newWithHooks(
+    GraphQLEnumType,
+    {
+      name: `Upsert${tableTypeName}DoUpdateField`,
+      description: `Switch to apply to a given field when resolving upsert conflicts`,
```

Comment:
Cool stuff! It may seem like nit/bikeshedding, but i've learned that a fair amount of folks do actually use this plugin these days, so I want to make sure we pick some _names_ that will feel ergonomic to all of our users. Can we chat on type names and key names?

`Do` is verby--which somewhat makes sense as we're talking about modifiers _during_ an update--however this type I think can be a bit more self descriptive, as it's not the verb/action itself but input to the action. I feel like the enum type semantically represents column upsert modifiers, and I'd like they typename to reflect that somehow. I'd go for something verbose such as `UpsertConflictCustomColumnProcessing` / `UpsertConflictColumnDirective`. What do you think? Something similar, perhaps?

Along the same lines, the `onConflict` clause could have something like the following, for more user friendly semantics?

```gql
onConflict: {
  columnDirectives: { foo: ignore, bar: current_timestamp } 
  // or
  inputOverrides: { ... }
  // or
  customColumnProcessing: { ... }
  // or
  // ...your ideas here
}
```

the essence that I feel is just slightly missing in the current state is that this special case handling doesn't communicate by context alone that it is a custom update mapping. to grok that, i had to read into the values in the examples. it was self evident immediately, but if someone is exploring or building up a mutation via graphiql, i think something like these names may enhance the discovery process.

super rambling response 😆 hopefully it was somewhat coherent. lemme know your thoughts

Reply:
i'll avoid commenting on the subsequent types in the PR until we chat further on ^^ :)

Reply:
Mmmm actually, the parity with the SQL does feel pretty good. I'm now just thinking "is there anything in the doUpdate block that we may want in the future that _isn't_ values like this enum". `do update set...` usually takes tuples of the fields to update in SQL, but here, we're semantically _deviating_ by accepting _customizations to `do update set ...`_ of which the plugin already provides, versus accepting raw KV pairs to `do update set ...`.

maybe...

```graphql
{
  onConflict: {
    doUpdate: {
      setColumn: { // maybe? follows PG grammar of `SET { column = ...` (https://www.postgresql.org/docs/9.1/sql-update.html)
         foo: current_timestamp
      }
    }
}
```

Reply:
(meant to make this comment last week but never clicked the button 🤦)

I see what you mean. Enabling the capability during plugin initialization is a start at preventing users' users from having too much power. My main goal was to provide as much flexibility as postgresql gives you, but all I really need for my current use case is covered by the `updateOnConflict` smart tag, which of course is defined at the schema level and not exposed to the user level. As far as how to name and structure things in my current implementation, that is up to you (if you don't want it ripped out), but I feel like it is concise enough and carries no uncovered side effects, even though it is more intuitive for those devs who have had experience with postgresql upserts.